.github/workflows/build-macos.yaml

```diff
@@ -2,6 +2,9 @@ name: Build on Mac OS
 run-name: Build on Mac OS
 
 on:
+  pull_request:
+    paths:
+      - '.github/workflows/build-macos.yaml'
```

Comment:
@reedloden I've added this temporarily here to see what happens with the macOS build.

Reply:
And it fails with a permission issue: https://github.com/gravitational/teleport/actions/runs/5469360603/jobs/9958206534?pr=28741.

@camscale Could you take a look?

Reply:
Ok. pushing a new commit to see what's going on